classes/MultiStoreFixer.php

```diff
@@ -0,0 +1,165 @@
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\PrestashopCheckout;
+
+class MultiStoreFixer
+{
+    const UNIQUE_CONFIGURATION_KEYS = [
+        'PS_CHECKOUT_SHOP_UUID_V4',
+        'PS_CHECKOUT_PAYPAL_ID_MERCHANT',
+        'PS_PSX_FIREBASE_ID_TOKEN',
+        'PS_PSX_FIREBASE_LOCAL_ID',
+        'PS_PSX_FIREBASE_REFRESH_TOKEN',
+    ];
+
+    /**
+     * @return bool
+     */
+    public function run()
+    {
+        return \Shop::isFeatureActive() ? $this->fixMultiStore() : $this->fixSingleStore();
+    }
+
+    /**
+     * In single store mode, we remove global value and save only value for default shop
+     *
+     * @return bool
+     */
+    private function fixSingleStore()
+    {
+        $result = true;
+
+        foreach (self::UNIQUE_CONFIGURATION_KEYS as $key) {
+            // Get current value
+            $value = \Configuration::get($key);
+
+            // Remove current global value
+            \Configuration::deleteByName($key);
+
+            // Save current value associated to default shop
+            $result &= \Configuration::updateValue(
+                $key,
+                $value,
+                false,
+                null,
+                (int) \Context::getContext()->shop->id
+            );
+        }
+
+        return (bool) $result;
+    }
+
+    /**
+     * In multi store mode, we need to find shop used for payment
+     * Please note Configuration::get return value for all shops or for group if no value for shop is found
+     *
+     * @return bool
+     */
+    private function fixMultiStore()
+    {
+        $result = true;
+        $currentValuesByShop = [];
+        $shopListUsedByOrder = $this->getShopListUsedByOrder();
+
+        foreach (self::UNIQUE_CONFIGURATION_KEYS as $key) {
+            $globalValueUsed = false;
+
+            // Get current values for each shop
+            foreach (\Shop::getShops(false, null, true) as $shopId) {
+                $globalValue = \Configuration::getGlobalValue($key);
+                $shopHasGlobalValue = false === empty($globalValue);
+                $shopHasUniqueKey = \Configuration::hasKey($key, null, null, (int) $shopId);
+                $shopValue = \Configuration::get($key, null, null, (int) $shopId);
+                $shopHasValue = false === empty($shopValue);
+                $isDefaultShop = (int) $shopId === (int) \Configuration::get('PS_SHOP_DEFAULT');
+
+                if ($shopHasUniqueKey && $shopHasValue) {
+                    // This shop has a unique value, we keep it
+                    $currentValuesByShop[(int) $shopId][$key] = $shopValue;
+                } elseif (false === $globalValueUsed && $shopHasValue && in_array($shopId, $shopListUsedByOrder)) {
+                    // This shop has order placed with checkout, we take global value found (can be value of group or all shop)
```

Comment:
Are you sure about this comment? I don't get the relation between the configuration data & an order.

Reply:
Yes, thanks to `in_array($shopId, $shopListUsedByOrder)` I check if this shop has been used to place an order.